### DISCUSSION TITLE
Foundry plugin – script deployment along with generate on watch feature

### DISCUSSION THREAD

**Discussion by dalechyn** [wevm/wagmi#1968] [Ideas]
Greetings.

I am confident that enabling `@wagmi/cli`'s foundry plugin to watch and re-deploy foundry scripts with saving the deployment addresses on the fly could be a potentially good and relatively easy feature to implement.

-----
#### Consider the next example:

I have a dApp and I want to enable the best DX. dApp is using some contracts that are deployed once on the chain (i.e. `Uniswap02Router` that we all know). But the development is rapid, and changing the address of the deployed contract every time I redeploy it locally may be a hard experience, as I would need to replace the address either on the frontend side via env files, or in the `wagmi.config.{js,ts}` in the `deployments` field.

-----
#### What we can do with it?
Since all deployments (including local) in Foundry are being saved in the `broadcast/{chainId}/run-latest.json` file, we can read the deployed contract addresses and inject them in the `wagmi.config.ts`.

For such a feature however, an external program is needed to be called (Foundry) and the `forge script --watch` must be implemented  (not implemented although that is defined in the docs https://github.com/foundry-rs/foundry/issues/4476), because the contracts have to be re-compiled and `@wagmi/cli` would have to regenerate the hooks **after** the contracts are redeployed (since those are two independent pieces of software they would have to communicate via IPC or some other interface).

That may be not feasible as of now, but maybe such an idea can be rethought to strive for new highs of dApp DX.
What do you think?

**Comment by tmm** [comment]
Hot Contract Redeployment (h/t @jxom) in development is definitely something we are interested in and on the roadmap! Busy at the moment preparing for `wagmi@1`, but anyone is welcome to play around with this in the meantime.
